lib/ABI/FunctionType/Support.cpp

```diff
@@ -0,0 +1,29 @@
+/// \file Support.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/ABI/FunctionType/Support.h"
+
+namespace abi::FunctionType {
+
+void replaceReferences(const model::Type::Key &OldKey,
+                       const model::TypePath &NewTypePath,
+                       TupleTree<model::Binary> &Model) {
+  auto Visitor = [&](model::TypePath &Visited) {
+    if (!Visited.isValid())
+      return; // Ignore empty references
+
+    model::Type *Current = Visited.get();
+    revng_assert(Current != nullptr);
+
+    if (Current->key() == OldKey)
+      Visited = NewTypePath;
+  };
+  Model.visitReferences(Visitor);
+  Model->Types().erase(OldKey);
```

Comment:
This is the old code (written before `TupleTree::replaceReferences` was a thing). Sorry if it confused you - it was just moved here so it can be shared once the file gets split.

Reply:
I'm confused, `replaceReferences` existed *before* this PR. Is there a reason not to use it?

Reply:
This code is not a part this PR, it's just relocated. Sorry for the confusion.

Reply:
> This code is not a part this PR

What do you mean? That it should be removed from this PR?

Reply:
It's on develop. This commit just moved it around.

Reply:
So we can drop it?

Reply:
It's reworked in a separate commit: https://github.com/revng/revng/pull/232/commits/bcf3f5587f9f7ac0bea9285cd283129fdd16c23c#diff-fdb1bb2652b4119c0c0784c0f1f53d3966e9a625beeac405ac5a8a3300681596L26